### PR TITLE
changed default num_points in OrbitPlotter class

### PR DESCRIPTION
**Pull request by nikita-astronaut**
Addresses #294 
I simple changed the default num_points in OrbitPlotter class, now orbits in the example look smooth.
Still, this is not the best solution. Nonsmoothness appears in the vicinity of perihelion, cause the speed there is the highest, but the points in time are chosen uniformly, so the difference in position is the largest. 

What I think could be done next is to place points non uniformly but with density ~1/current_speed, so that between every two points the position change is the same.

What do you think?

P.S. I am actually looking to joining your project later in the GSoC18.

### CODE REVIEW

**Comment by Juanlu001** [comment]
Hi @nikita-astronaut, thanks for your interest in poliastro!

As you say, just increasing the number of points is not the best solution. I added more information in the original issue, so please take your time to read it.

**Comment by nikita-astronaut** [comment]
Hi @Juanlu001 
I was surprised (but why?) that the functions of transitions between anomalies are already present in the package, so I used them. Now the angle is sampled uniformly from eccentric anomaly, not from the true anomaly. Seems like now the orbit looks smooth for only `num_points = 150` instead of `num_points = 300` I used previously.

before
after_1
after_2

I attached three images. The first is how it was before (true anomaly + 100 points), second: eccentric anomaly + 100 points (the problem is ameliorated but nonsmoothness however is still seen), third: eccentric anomaly + 150 points (orbit looks smooth).

Now, I think, I'll try to go further and work with s-integrator you've mentioned.

File: docs/source/examples/Using NEOS package.ipynb [hunk 1]
```diff
@@ -29,7 +29,9 @@
   {
    "cell_type": "code",
    "execution_count": 1,
-   "metadata": {},
+   "metadata": {
```
Comment (Juanlu001) [inline docs/source/examples/Using NEOS package.ipynb]: Can you restore the changes in this notebook?

File: src/poliastro/plotting.py [hunk 2]
```diff
@@ -59,15 +59,15 @@ class OrbitPlotter(object):
...
+    def __init__(self, ax=None, num_points=150):
         """Constructor.
 
         Parameters
         ----------
         ax : ~matplotlib.axes.Axes
             Axes in which to plot. If not given, new ones will be created.
         num_points : int, optional
-            Number of points to use in plots, default to 100.
+            Number of points to use in plots, default to 300.
```
Comment (Juanlu001) [inline src/poliastro/plotting.py]: Update docstring :)

File: src/poliastro/twobody/orbit.py [hunk 3]
```diff
@@ -4,10 +4,13 @@
 
 from astropy import units as u
 from astropy import time
+
 from astropy.coordinates import CartesianRepresentation, get_body_barycentric_posvel
 
 from poliastro.constants import J2000
 from poliastro.twobody.angles import nu_to_M
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: What about importing all these functions in the same line?

File: src/poliastro/twobody/orbit.py [hunk 4]
```diff
@@ -294,12 +297,17 @@ def sample(self, values=None, function=propagate):
 
         elif isinstance(values, int):
             if self.ecc < 1:
-                nu_values = np.linspace(0, 2 * np.pi, values) * u.rad
+                # first sample eccenric anomaly, then transform into true anomaly
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: s/eccenric/eccentric/g

Also, what about adding a reference to the report in the comment? "Why sample uniformly on the eccentric anomaly to minimize error in the apocenter, see ..."

File: src/poliastro/twobody/orbit.py [hunk 5]
```diff
@@ -294,12 +297,17 @@ def sample(self, values=None, function=propagate):
...
+                E_values = np.linspace(0, 2 * np.pi, values) * u.rad
+                nu_values = E_to_nu(E_values, self.ecc)
+                
             else:
                 # Select a sensible limiting value for non-closed orbits
                 # This corresponds to r = 3p
                 nu_limit = np.arccos(-(1 - 1 / 3.) / self.ecc)
-                nu_values = np.linspace(-nu_limit, nu_limit, values)
+                E_limit = nu_to_E(nu_limit, self.ecc) / u.rad
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: Not sure if the eccentric anomaly makes sense in the context of hyperbolic orbits. I wouldn't be surprised if this fails.

**Comment by nikita-astronaut** [comment]
I think I fixed everything you were asking)

**Comment by Juanlu001** [comment]
Awesome! Thanks for your contribution :grin:

**Pull request merged**